### PR TITLE
VEN-942 | Fix the bug of refreshing paginated tables & upgrade 'react-table'

### PR DESCRIPTION
**Pull request by Nizar-Rahme**
## Description :sparkles:
- Upgrade `react-table`
- Prevent side effects when the initial `sortBy` matches the current state

### Closes :no_good_woman:
[**VEN-942:**](https://helsinkisolutionoffice.atlassian.net/browse/VEN-942) Refreshing paginated tables go back to the first page

### Automated tests :gear:️
- Fixed all relevant snapshots

### Manual testing :construction_worker_man:
- Go to one of the backend paginated tables e.g. Applications List
- Go to the second page
- Refresh the page
- The table should stay and render the second page as expected

### CODE REVIEW

File: src/common/table/Table.tsx [hunk 1]
```diff
@@ -253,9 +253,12 @@ const Table = <D extends { id: string }>({
     gotoPage(0);
   }, [gotoPage, state.sortBy, state.filters, state.globalFilter]);
 
+  const initialSortBy = initialState?.sortBy;
+
   useEffect(() => {
+    if (initialSortBy?.[0]?.id === state.sortBy[0]?.id && initialSortBy?.[0]?.desc === state.sortBy[0]?.desc) return;
```
Comment (TatuArvela) [inline src/common/table/Table.tsx:259]: This is a bit complicated, could this be expressed a bit more clearly?

File: src/common/table/__tests__/__snapshots__/Table.test.jsx.snap [hunk 2]
```diff
@@ -8,7 +8,7 @@ exports[`Table renders normally 1`] = `
     <div
       class="table primary noMainHeader"
       role="table"
-      style="min-width:450px"
+      style="min-width:0px"
```
Comment (TatuArvela) [inline src/common/table/__tests__/__snapshots__/Table.test.jsx.snap:11]: This makes tables that overflow horizontally now display incorrectly

**Review by TatuArvela — CHANGES_REQUESTED** [review]
There's now a problem with table sizing

File: src/common/table/__tests__/__snapshots__/Table.test.jsx.snap (hunk 2 above)
Comment (Nizar-Rahme) [reply src/common/table/__tests__/__snapshots__/Table.test.jsx.snap:11]: I'll check if I can control it, I got these changes by merely upgrading the library.

Comment (Nizar-Rahme) [reply src/common/table/__tests__/__snapshots__/Table.test.jsx.snap:11]: So this was the culprit; they [changed the logic of calculating `min-width`](https://github.com/tannerlinsley/react-table/pull/2639/files) in `react-table` v `7.6.0`.

Comment (Nizar-Rahme) [reply src/common/table/__tests__/__snapshots__/Table.test.jsx.snap:11]: Therefore, we have to add `minWidth` to the columns that use `width` property. 
I added them to all the tables that have `width` already set.

File: src/common/table/Table.tsx (hunk 1 above)
Comment (Nizar-Rahme) [reply src/common/table/Table.tsx]: Done.

File: src/common/table/__tests__/__snapshots__/Table.test.jsx.snap (hunk 2 above)
Comment (TatuArvela) [reply src/common/table/__tests__/__snapshots__/Table.test.jsx.snap:11]: Great 👍 Is `width` now needed for those columns?

Comment (TatuArvela) [reply src/common/table/__tests__/__snapshots__/Table.test.jsx.snap:11]: I tried it out, and it turns out it is. This is kind of messy, but let's investigate a better solution later.

**Review by TatuArvela — APPROVED** [review]
LGTM 👍 Rebased it

**Pull request merged**